controllers/clustergroupupgrade_controller.go

```diff
@@ -1448,16 +1522,43 @@ func (r *ClusterGroupUpgradeReconciler) validateCR(ctx context.Context, clusterG
 		}, foundManagedCluster)
 
 		if err != nil {
-			return fmt.Errorf("Cluster %s is not a ManagedCluster", cluster)
+			return reconcile, fmt.Errorf("Cluster %s is not a ManagedCluster", cluster)
```

Comment:
Yes, this would be good to have, but I would add it to the same list of enhancements as there is also the question of what status/action we want to take if such checks fail.